### PR TITLE
Week 4 user authorization

### PR DESCRIPTION
**Pull request by mahimashanware**
some overlap with adding a map (by accident)

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -37,18 +37,20 @@
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments = new ArrayList<>(); 
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private UserService userService = UserServiceFactory.getUserService();
+  private int maxComments = 10;
```
Comment (jonathanjlin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Looks like this was removed in #8, why was this added back in?

**Review by jonathanjlin — COMMENTED** [review]
It looks like there's some changes from previous weeks like the map and posting comments parts, can you update the PR branch so that the PR only has the authorization changes in it?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (mahimashanware) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I realized I didn't need it then, but ended up needing it in this week's content.

Comment (jonathanjlin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The same comment applies still, requests that specify a max-comments query parameter will affect subsequent requests without the query parameter specified

**Review by jonathanjlin — APPROVED** [review]
